nomad/leader.go

```diff
@@ -166,6 +167,17 @@ func (s *Server) establishLeadership(stopCh chan struct{}) error {
 		s.logger.Printf("[ERR] nomad: heartbeat timer setup failed: %v", err)
 		return err
 	}
+
+	// COMPAT 0.4 - 0.4.1
+	// Reconcile the summaries of the registered jobs. We reconcile summaries
+	// only if the server is 0.4.1 since summaries are not present in 0.4 they
+	// might be incorrect after upgrading to 0.4.1 the summaries might not be
+	// correct
+	if strings.HasPrefix(s.config.Build, "0.4.1") {
```

Comment:
We can update this check when we do 0.5.0 and check if the version is either 0.4.1 or 0.5.0